pkg/providers/imagefamily/bootstrap/aksbootstrap_test.go

```diff
@@ -58,3 +65,204 @@ func TestKubeBinaryURL(t *testing.T) {
 		})
 	}
 }
+
+func TestAKS_aksBootstrapScript(t *testing.T) {
```

Comment:
Can we test that the values for CSE are set correctly [here](https://github.com/Azure/karpenter-provider-azure/blob/main/pkg/providers/instancetype/suite_test.go#L541) instead using e2e unit tests? This will test that all of the expected configuration from karpenter is resolved correctly and other layers than just purely the BootstrapScript. 

A test here validating the customData we set is correct would be good. 


Also can we not use go tests? We are looking to move away from them in favor of consistent ginkgo testing.

Reply:
Sure.

> Also can we not use go tests? We are looking to move away from them in favor of consistent ginkgo testing

Was not aware of this but yes, I can change them.